x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/processors/index.tsx

```diff
@@ -44,11 +44,13 @@ import { UseProcessingSimulatorReturn } from '../hooks/use_processing_simulator'
 export interface ProcessorPanelProps {
   definition: IngestStreamGetResponse;
   onWatchProcessor: UseProcessingSimulatorReturn['watchProcessor'];
+  refreshSimulation: UseProcessingSimulatorReturn['refreshSimulation'];
 }
 
 export interface AddProcessorPanelProps extends ProcessorPanelProps {
   isInitiallyOpen?: boolean;
   onAddProcessor: UseDefinitionReturn['addProcessor'];
+  samples: RecursiveRecord[];
```

Comment:
thoughts about making this an explicit type like `SampleDocument` so we don't start using this type in any place where it's easy? it's essentially `any`

Reply:
> it's essentially any

What do you mean by this? The type is for typing nested structures with some limitations (we don't expect arrays of objects, hence that's not allowed) and we could get nested/flattened objects from the sampling. 
Do you have anything specific in mind for the `SampleDocument` type?

Reply:
not sure if I follow, where would you keep it then?

Reply:
What I mean is that we use the `RecursiveRecord` type might be used in other places (such as the simulation API) to type docs `_source` without them necessarily represent sample documents. I guess it would be confusing to use the SampleDocument type for something that it's not used as a sample.

Reply:
I will move the `RecursiveRecord` type and schema to `kbn-schema`, then rename all the usages to `SampleDocument`. Once we need `RecursiveRecord` somewhere else, we can easily split it out as a standalone thing, but let's not overcomplicate from the start.